### PR TITLE
e2e revisions of chapter 8 code

### PR DESCRIPTION
**Pull request by jwills**
@srowen for review when you get a chance.

### CODE REVIEW

File: ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala [hunk 1]
```diff
@@ -6,148 +6,104 @@
...
       case t if t.isLeft => t.left.get
-    })
+    }).toDS
     taxiGood.cache()
 
-    def hours(trip: Trip): Long = {
-      val d = new Duration(trip.pickupTime, trip.dropoffTime)
-      d.getStandardHours
+    val hours = (pickup: Long, dropoff: Long) => {
+      (dropoff - pickup) / (1000 * 60 * 60)
```
Comment (srowen) [inline ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala]: Not sure if it's actually clearer, but there's `TimeUnit.HOURS.convert(dropoff - pickup, TimeUnit.MILLISECONDS)`

File: ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala [hunk 2]
```diff
@@ -6,148 +6,104 @@
...
-    }
+    // register the UDF, use it in a where clause
+    spark.udf.register("hours", hours)
+    val taxiClean = taxiGood.where("hours(pickupTime, dropoffTime) BETWEEN 0 AND 3")
 
-    val geojson = scala.io.Source.fromURL(getClass.getResource("/nyc-boroughs.geojson")).mkString
+    val geojson = scala.io.Source.fromURL(this.getClass.getResource("/nyc-boroughs.geojson")).mkString
 
     val features = geojson.parseJson.convertTo[FeatureCollection]
     val areaSortedFeatures = features.sortBy(f => {
```
Comment (srowen) [inline ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala]: Aside, really minor: consider `.sortBy { f =>` as slightly more idiomatic, and one less paren

File: ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala [hunk 3]
```diff
@@ -6,148 +6,104 @@
...
-
-    val taxiBad = taxiParsed.collect({
-      case t if t.isRight => t.right.get
-    })
-    taxiBad.collect().foreach(println)
+    val spark = SparkSession.builder().getOrCreate()
+    import spark.implicits._
 
+    val taxiRaw = spark.read.option("header", "true").csv("taxidata")
+    val taxiParsed = taxiRaw.rdd.map(safe(parse))
```
Comment (srowen) [inline ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala]: If you just use a Dataset here, do you run into problems with it not knowing what to do with `Either`? I've always felt a little bit funny about `Either` but it's not deprecated and a whole little case class seems like overkill. Using Dataset would be nice for the clean sweep but not that important IMHO if it's used as some teaching moment in the text.

File: ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala (hunk 2 above)
Comment (jwills) [inline ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala]: Ack, will do.

File: ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala (hunk 3 above)
Comment (jwills) [inline ch08-geotime/src/main/scala/com/cloudera/datascience/geotime/RunGeoTime.scala]: (FWIW, the data from ch02 parses cleanly using the spark-csv typed Dataset API, so I'm going to use the easy, idiomatic pattern in that chapter.)

**Pull request merged**